cmake/stm32/devices.cmake

```diff
@@ -1125,9 +1128,15 @@ function(stm32_get_devices_by_family DEVICES)
     if(ARG_FAMILY)
         list(FILTER LIST INCLUDE REGEX "^${ARG_FAMILY}")
     endif()
+    if(NOT LIST)
```

Comment:
This bit is weird, where does STM32_ALL_DEVICES come from? this test checks this var, not the `LIST` var.
I also strongly advice against using `LIST` as a var name. It's not self explanatory, and it is also a cmake key word!

Reply:
> This bit is weird, where does STM32_ALL_DEVICES come from ?

This is one of the main problems for me. It comes from here : https://github.com/ObKo/stm32-cmake/blob/master/cmake/stm32/devices.cmake
and is used only for these functions. See my very first comment in the PR where I already mentioned it and @rmspacefish seems to have an use the function.